<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>StreamSettingsWidget</class>
 <widget class="QWidget" name="StreamSettingsWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>455</width>
    <height>379</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Stream Settings Widget</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>Protocol Settings</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="1" column="0">
        <widget class="QStackedWidget" name="v2rayStackView">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>1</width>
           <height>0</height>
          </size>
         </property>
         <property name="currentIndex">
          <number>0</number>
         </property>
         <widget class="QWidget" name="tcpStackPage">
          <layout class="QGridLayout" name="gridLayout_2" rowstretch="0,0,1,0">
           <item row="0" column="0" colspan="4">
            <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,1">
             <item>
              <widget class="QLabel" name="label">
               <property name="text">
                <string>Type</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="tcpHeaderTypeCB">
               <item>
                <property name="text">
                 <string notr="true">none</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string notr="true">http</string>
                </property>
               </item>
              </widget>
             </item>
            </layout>
           </item>
           <item row="2" column="0" colspan="2">
            <widget class="QPlainTextEdit" name="tcpRequestTxt">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="lineWrapMode">
              <enum>QPlainTextEdit::NoWrap</enum>
             </property>
             <property name="readOnly">
              <bool>true</bool>
             </property>
             <property name="plainText">
              <string/>
             </property>
             <property name="backgroundVisible">
              <bool>false</bool>
             </property>
             <property name="placeholderText">
              <string notr="true"/>
             </property>
            </widget>
           </item>
           <item row="2" column="2" colspan="2">
            <widget class="QPlainTextEdit" name="tcpRespTxt">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="lineWrapMode">
              <enum>QPlainTextEdit::NoWrap</enum>
             </property>
             <property name="readOnly">
              <bool>true</bool>
             </property>
             <property name="plainText">
              <string/>
             </property>
             <property name="placeholderText">
              <string notr="true"/>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QPushButton" name="tcpRequestEditBtn">
             <property name="text">
              <string>Edit</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QPushButton" name="tcpRequestDefBtn">
             <property name="text">
              <string>Reset</string>
             </property>
            </widget>
           </item>
           <item row="3" column="2">
            <widget class="QPushButton" name="tcpResponseEditBtn">
             <property name="text">
              <string>Edit</string>
             </property>
            </widget>
           </item>
           <item row="3" column="3">
            <widget class="QPushButton" name="tcpRespDefBtn">
             <property name="text">
              <string>Reset</string>
             </property>
            </widget>
           </item>
           <item row="1" column="2" colspan="2">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Response</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0" colspan="2">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Request</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="httpStackPage">
          <layout class="QGridLayout" name="gridLayout_8" rowstretch="0,0,0,1">
           <item row="0" column="0">
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>Path</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLineEdit" name="httpPathTxt">
             <property name="placeholderText">
              <string notr="true">/</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Host</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QPlainTextEdit" name="httpHostTxt">
             <property name="placeholderText">
              <string notr="true">myhost.mydomain.com</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="wsStackPage">
          <layout class="QGridLayout" name="gridLayout_9" rowstretch="0,0,1">
           <item row="0" column="0">
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <widget class="QLabel" name="label_7">
               <property name="text">
                <string>Path</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="wsPathTxt">
               <property name="text">
                <string/>
               </property>
               <property name="placeholderText">
                <string notr="true">/wsPath</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="2" column="0">
            <widget class="QPlainTextEdit" name="wsHeadersTxt">
             <property name="placeholderText">
              <string notr="true">Key|Value</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_26">
             <property name="text">
              <string>Headers</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="mKCPStackPage">
          <layout class="QFormLayout" name="formLayout_5">
           <item row="0" column="0">
            <widget class="QLabel" name="label_16">
             <property name="text">
              <string>Header</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QComboBox" name="kcpHeaderType">
             <property name="sizePolicy">
              <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <item>
              <property name="text">
               <string notr="true">none</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string notr="true">srtp</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string notr="true">utp</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string notr="true">wechat-video</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string notr="true">dtls</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string notr="true">wireguard</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_13">
             <property name="text">
              <string>Seed</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLineEdit" name="kcpSeedTxt"/>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_23">
             <property name="text">
              <string>Capacity</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="1,1">
             <item>
              <widget class="QSpinBox" name="kcpUploadCapacSB">
               <property name="suffix">
                <string notr="true"> MB/s</string>
               </property>
               <property name="prefix">
                <string>Up: </string>
               </property>
               <property name="maximum">
                <number>4096</number>
               </property>
               <property name="value">
                <number>5</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="kcpDownCapacitySB">
               <property name="minimumSize">
                <size>
                 <width>80</width>
                 <height>0</height>
                </size>
               </property>
               <property name="suffix">
                <string notr="true"> MB/s</string>
               </property>
               <property name="prefix">
                <string>Down: </string>
               </property>
               <property name="maximum">
                <number>4096</number>
               </property>
               <property name="value">
                <number>20</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="label_22">
             <property name="text">
              <string>Buffer Size</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <layout class="QHBoxLayout" name="horizontalLayout_6" stretch="1,1">
             <item>
              <widget class="QSpinBox" name="kcpWriteBufferSB">
               <property name="suffix">
                <string notr="true"> MB</string>
               </property>
               <property name="prefix">
                <string>Write Size: </string>
               </property>
               <property name="maximum">
                <number>1024</number>
               </property>
               <property name="value">
                <number>2</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="kcpReadBufferSB">
               <property name="suffix">
                <string notr="true"> MB</string>
               </property>
               <property name="prefix">
                <string>Read Size: </string>
               </property>
               <property name="maximum">
                <number>1024</number>
               </property>
               <property name="value">
                <number>2</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="label_24">
             <property name="text">
              <string>Congestion</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QCheckBox" name="kcpCongestionCB">
             <property name="text">
              <string>Congestion</string>
             </property>
            </widget>
           </item>
           <item row="5" column="0">
            <widget class="QLabel" name="label_15">
             <property name="text">
              <string>Options</string>
             </property>
            </widget>
           </item>
           <item row="5" column="1">
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QLabel" name="label_8">
               <property name="text">
                <string>MTU</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="kcpMTU">
               <property name="sizePolicy">
                <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="suffix">
                <string notr="true"> B</string>
               </property>
               <property name="minimum">
                <number>576</number>
               </property>
               <property name="maximum">
                <number>1460</number>
               </property>
               <property name="value">
                <number>1350</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_10">
               <property name="text">
                <string>TTI</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="kcpTTI">
               <property name="sizePolicy">
                <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="suffix">
                <string notr="true"> ms</string>
               </property>
               <property name="minimum">
                <number>10</number>
               </property>
               <property name="maximum">
                <number>100</number>
               </property>
               <property name="value">
                <number>50</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="dsStackPage">
          <layout class="QFormLayout" name="formLayout_2">
           <item row="0" column="0">
            <widget class="QLabel" name="label_17">
             <property name="text">
              <string>Path</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLineEdit" name="dsPathTxt">
             <property name="text">
              <string/>
             </property>
             <property name="placeholderText">
              <string notr="true">/dsPath</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="quicStackPage">
          <layout class="QFormLayout" name="formLayout_3">
           <item row="0" column="0">
            <widget class="QLabel" name="label_18">
             <property name="text">
              <string>Security</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QComboBox" name="quicSecurityCB">
             <item>
              <property name="text">
               <string notr="true">none</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string notr="true">aes-128-gcm</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string notr="true">chacha20-poly1305</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_19">
             <property name="text">
              <string>Key</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLineEdit" name="quicKeyTxt">
             <property name="text">
              <string/>
             </property>
             <property name="placeholderText">
              <string>keys</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_20">
             <property name="text">
              <string>Headers</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QComboBox" name="quicHeaderTypeCB">
             <item>
              <property name="text">
               <string notr="true">none</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string notr="true">srtp</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string notr="true">utp</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string notr="true">wechat-video</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string notr="true">dtls</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string notr="true">wireguard</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="3" column="0">
            <spacer name="verticalSpacer_2">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
       <item row="0" column="0">
        <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="0,1">
         <item>
          <widget class="QLabel" name="transportLabel">
           <property name="text">
            <string>Transport Protocol</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="transportCombo">
           <item>
            <property name="text">
             <string notr="true">tcp</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string notr="true">http</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string notr="true">ws</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string notr="true">kcp</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string notr="true">domainsocket</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string notr="true">quic</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>TLS Settings</string>
      </attribute>
      <layout class="QFormLayout" name="formLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="label_21">
         <property name="text">
          <string>Security Type</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QComboBox" name="securityTypeCB">
         <item>
          <property name="text">
           <string notr="true">None</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string notr="true">TLS</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string notr="true">XTLS</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="1" column="0" colspan="2">
        <widget class="QCheckBox" name="allowInsecureCB">
         <property name="text">
          <string>Allow Insecure Certificates</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0" colspan="2">
        <widget class="QCheckBox" name="disableSessionResumptionCB">
         <property name="text">
          <string>Disable Session Resumption</string>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>Server Address (SNI)</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QLineEdit" name="serverNameTxt"/>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="label_14">
         <property name="text">
          <string>ALPN</string>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <widget class="QLineEdit" name="alpnTxt"/>
       </item>
       <item row="3" column="0" colspan="2">
        <widget class="QCheckBox" name="disableSystemRoot">
         <property name="text">
          <string>Disable System Root Certificates</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>SOCK Options</string>
      </attribute>
      <layout class="QFormLayout" name="formLayout_4">
       <item row="0" column="0">
        <widget class="QLabel" name="label_28">
         <property name="text">
          <string>Mark</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSpinBox" name="soMarkSpinBox"/>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_29">
         <property name="text">
          <string>TCP Fast Open</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QCheckBox" name="tcpFastOpenCB">
         <property name="text">
          <string>Enabled</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_27">
         <property name="text">
          <string>tProxy Mode</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QComboBox" name="tProxyCB">
         <item>
          <property name="text">
           <string notr="true">off</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string notr="true">redirect</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string notr="true">tproxy</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>tabWidget</tabstop>
  <tabstop>transportCombo</tabstop>
  <tabstop>tcpHeaderTypeCB</tabstop>
  <tabstop>tcpRequestTxt</tabstop>
  <tabstop>tcpRespTxt</tabstop>
  <tabstop>tcpRequestEditBtn</tabstop>
  <tabstop>tcpRequestDefBtn</tabstop>
  <tabstop>tcpResponseEditBtn</tabstop>
  <tabstop>tcpRespDefBtn</tabstop>
  <tabstop>httpPathTxt</tabstop>
  <tabstop>httpHostTxt</tabstop>
  <tabstop>wsPathTxt</tabstop>
  <tabstop>wsHeadersTxt</tabstop>
  <tabstop>kcpUploadCapacSB</tabstop>
  <tabstop>kcpDownCapacitySB</tabstop>
  <tabstop>dsPathTxt</tabstop>
  <tabstop>quicSecurityCB</tabstop>
  <tabstop>quicKeyTxt</tabstop>
  <tabstop>quicHeaderTypeCB</tabstop>
  <tabstop>allowInsecureCB</tabstop>
  <tabstop>serverNameTxt</tabstop>
  <tabstop>soMarkSpinBox</tabstop>
  <tabstop>tcpFastOpenCB</tabstop>
  <tabstop>tProxyCB</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
